clusterpools/apply.sh

```diff
@@ -740,6 +791,10 @@ fi
 if [[ "$CLUSTERPOOL_SKIP_MACHINEPOOL" == "true" ]]; then
     printf "\n  skipMachinePools: True" >> ./${CLUSTERPOOL_NAME}/${CLUSTERPOOL_NAME}.clusterpool.yaml
 fi
+# add a ManagedClusterSet if specified
+if [[ "$MANAGEDCLUSTERSET_NAME" ]]; then
+    MANAGEDCLUSTERSET_NAME="'cluster.open-cluster-management.io/clusterset: $MANAGEDCLUSTERSET_NAME'" yq e '.metadata.labels = [env(MANAGEDCLUSTERSET_NAME)]' -i ./${CLUSTERPOOL_NAME}/${CLUSTERPOOL_NAME}.clusterpool.yaml
```

Comment:
I get an error here:
```
error: unable to decode "./test-pool/test-pool.clusterpool.yaml": resource.metadataOnlyObject.ObjectMeta: v1.ObjectMeta.Labels: ReadMapCB: expect { or n, but found [, error found in #10 byte of ...|"labels":["cluster.o|..., bigger context ...|io/v1","kind":"ClusterPool","metadata":{"labels":["cluster.open-cluster-management.io/clusterset: ac|...
```
Maybe some missing quotes around the key/values?

Reply:
Ah, no--it's expecting key/values. Not an array...

Reply:
Hmmmm - I tested this locally successfully - what yq version are you running @dhaiducek ?

Reply:
We've located and patched the issue.  Turns out yq 4.4 and 4.11 handle key-value pair creation differently (is my guess, I've upgraded and patched and validated on yq 4.12).